pkg/util/pki/csr.go

```diff
@@ -298,6 +306,27 @@ func buildKeyUsagesExtensionsForCertificate(crt *v1.Certificate) ([]pkix.Extensi
 	return extraExtensions, nil
 }
 
+func buildBasicConstraintsExtensionsForCertificate(isCA bool) (pkix.Extension, error) {
+
+	basicConstraints := pkix.Extension{
+		Id: OIDExtensionBasicConstraints,
+	}
+
+	constraint := struct {
+		IsCA bool
+	}{
+		IsCA: isCA,
+	}
+
+	var err error
+	basicConstraints.Value, err = asn1.Marshal(constraint)
```

Comment:
This code is missing to set Critical to true due to which some of the CAs won't issue certificate. 
The right code is to add 

    basicConstraints := pkix.Extension{
               Id: OIDExtensionBasicConstraints,
               Id:       OIDExtensionBasicConstraints,
               Critical: isCA,
        }